code/src/ng_link/ng_state.py

```diff
@@ -0,0 +1,348 @@
+from typing import List, Any, Optional, Union
+from .ng_layer import NgLayer
+from pint import UnitRegistry
+from pathlib import Path
+import sys
+import re
+from utils import utils
+
+# IO types
+PathLike = Union[str, Path]
+
+class NgState():
+    def __init__(self, input_config:dict, output_json:PathLike, verbose:Optional[bool]=False) -> None:
+        """
+        Class constructor
+        
+        Parameters
+        ------------------------
+        image_config: dict
+            Dictionary with the image configuration based on neuroglancer documentation.
+        output_json: PathLike
+            Path where the json will be written.
+        
+        verbose: Optional[bool]
+            If true, additional information will be shown. Default False.
+        """
+        
+        self.input_config = input_config
+        self.output_json = Path(self.__fix_output_json_path(output_json))
+        self.verbose = verbose
+        
+        # State and layers attributes
+        self.__state = {}
+        self.dimensions = {}
+        self.__layers = []
+        
+        self.__prefix = 'ng_link_'
+        
+        # Initialize principal attributes
+        self.initialize_attributes(self.input_config)
+    
+    def __fix_output_json_path(self, output_json:PathLike) -> str:
+        
+        """
+        Fixes the json output path in order to have a similar structure for all links.
+        
+        Parameters
+        ------------------------
+        output_json: PathLike
+            Path of the json output path.
+        
+        Returns
+        ------------------------
+        str
+            String with the fixed outputh path.
+        """
+        output_json = Path(
+            str(output_json).replace(
+                '/home/jupyter/', ''
+            ).replace(
+                "////", "//"
+            )
+        )
+        name = str(output_json.name)
+        
+        if not name.endswith('.json'):

```

Comment:
Munging file names dynamically makes me nervous.  Can we validate instead?

Reply:
I will work on that